GH-73991: Rework `pathlib.Path.copytree()` into `copy()` (#122369)
python/cpython: Doc/library/pathlib.rst

Copying, renaming and deleting
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. method:: Path.copy(target, *, follow_symlinks=True, dirs_exist_ok=False, \
                      preserve_metadata=False, ignore=None, on_error=None)

   Copy this file or directory tree to the given *target*.

   If the source is a file, the target will be replaced if it is an existing
   file. If the source is a symlink and *follow_symlinks* is true (the
   default), the symlink's target is copied. Otherwise, the symlink is
   recreated at the destination.

   If the source is a directory and *dirs_exist_ok* is false (the default), a
   :exc:`FileExistsError` is raised if the target is an existing directory.
   If *dirs_exists_ok* is true, the copying operation will continue if it

>    encounters existing directories, and files within the destination tree will

Reviewer: This sentence feels a bit longer than necessary. Maybe "Otherwise, the files within..." or "If *dirs_exists_ok* is true, the files within..."

Author: I think it's a tiny bit ambiguous if we leave out "the copying operating will continue _if it encounters existing directories_", because the bit I've italicised isn't part of the main condition in the first sentence, so it won't necessarily be assumed by the reader here.

Reviewer 2: The current wording seems fine. To be honest I'm not sure I understand the point of the condition (*if it encounters existing directories*), because surely this follows from not failing if a directory is encountered. So you could have

```
   If *dirs_exists_ok* is true, the copying operation will overwrite
   existing files within the destination tree with corresponding files
   from the source tree.
```

Copying, renaming and deleting
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. method:: Path.copy(target, *, follow_symlinks=True, \
                      preserve_metadata=False, dirs_exist_ok=False, \
                      ignore=None, on_error=None)

   Copy this file or directory tree to the given *target*. If both paths are

>    existing files, the target file will be replaced.

Reviewer: The path we're copying should always exist, maybe something like "When copying a file, *target* will be replaced if it is an existing file."?

>    If a symlink is encountered and *follow_symlinks* is true (the default),

Reviewer: This could be clearer for when copying a symlink, maybe "For a symlink if *follow_symlinks* is true..."?

   the symlink's target is copied. Otherwise, the symlink is recreated at the
   destination.

   If *preserve_metadata* is false (the default), only directory structures
   and file data are guaranteed to be copied. Set *preserve_metadata* to true
   to ensure that file and directory permissions, flags, last access and
   modification times, and extended attributes are copied where supported.
   This argument has no effect when copying files on Windows (metadata is
   always preserved in this case.)

>    If the source and target are existing directories and *dirs_exist_ok* is

Reviewer: "When copying a directory and target is an existing directory ..."

   to ensure that file and directory permissions, flags, last access and
   modification times, and extended attributes are copied where supported.

>    This argument has no effect when copying files on Windows (metadata is
>    always preserved in this case.)

Reviewer 2: Full stop outside of the brackets (and a wording suggestion)
```suggestion
   This argument has no effect when copying files on Windows (where 
   metadata is always preserved).
```
